ui/src/lib/yjs/useYNode/utils.ts

```diff
@@ -0,0 +1,235 @@
+import type { Node } from "@flow/types";
+
+import {
+  reassembleEdge,
+  yEdgeConstructor,
+  yNodeConstructor,
+} from "../conversions";
+import type { YEdge, YEdgesArray, YNodesArray, YWorkflow } from "../types";
+
+export function updateParentYWorkflowNode(
+  currentWorkflowId: string,
+  yParentWorkflow: YWorkflow,
+  node: Node,
+  params: any,
+) {
+  const yParentNodes = yParentWorkflow?.get("nodes") as YNodesArray | undefined;
+  if (!yParentNodes) return;
+
+  const parentNodes = yParentNodes.toJSON() as Node[];
+
+  // Update the subworkflow node with the updated input/output
+  const subworkflowNode = parentNodes.find((n) => n.id === currentWorkflowId);
+  if (!subworkflowNode) return;
+
+  const newPseudoPort = {
+    nodeId: node.id,
+    portName: params.routingPort,
+  };
+
+  const isRouterInput = node.data.outputs?.length;
+  const isRouterOutput = node.data.inputs?.length;
+
+  // Update pseudo ports and assign them back to the node
+  const updatedPseudoPorts = isRouterInput
+    ? updatePseudoPorts(
+        subworkflowNode.data.pseudoInputs ?? [],
+        newPseudoPort,
+        currentWorkflowId,
+        yParentWorkflow,
+        "target",
+      )
+    : isRouterOutput
+      ? updatePseudoPorts(
+          subworkflowNode.data.pseudoOutputs ?? [],
+          newPseudoPort,
+          currentWorkflowId,
+          yParentWorkflow,
+          "source",
+        )
+      : null;
+
+  if (!updatedPseudoPorts) return;
+
+  const updatedSubworkflowNode: Node = {
+    ...subworkflowNode,
+    data: {
+      ...subworkflowNode.data,
+      [isRouterInput ? "pseudoInputs" : "pseudoOutputs"]: updatedPseudoPorts,
+    },
+  };
+
+  const newParentNodes = parentNodes.map((node) =>
+    node.id === updatedSubworkflowNode.id ? updatedSubworkflowNode : node,
+  );
+
+  const newParentYNodes = newParentNodes.map((node) => yNodeConstructor(node));
+
+  yParentNodes.delete(0, parentNodes.length);
+  yParentNodes.insert(0, newParentYNodes);
+}
+
+// Function to update pseudoInputs or pseudoOutputs
+function updatePseudoPorts(
+  pseudoPorts: { nodeId: string; portName: string }[],
+  newPseudoPort: { nodeId: string; portName: string },
+  currentWorkflowId: string,
+  yParentWorkflow: YWorkflow,
+  edgeType: "source" | "target",
+) {
+  const portIndex = pseudoPorts.findIndex(
+    (port) => port.nodeId === newPseudoPort.nodeId,
+  );
+  const prevPseudoPort = portIndex !== -1 ? pseudoPorts[portIndex] : null;
```

Comment:
Pretty sure you just need to do `prevPseudoPort = pseudoPorts.find(
    (port) => port.nodeId === newPseudoPort.nodeId,
  );`
and use `prevPseudoPort` rather than the portIndex below.